_topics/en/soft-fork-activation.md

```diff
@@ -0,0 +1,438 @@
+    activation.  Additionally, not knowing the exact activation time in
+    advance meant it was difficult for protocol developers, merchant
+    system administrators, and mining pool operators to be available in
+    the hours shortly after activation occurred in case there were any
+    problems that needed a quick response.
+
+  BIP9 versionbits attempted to solve these problems by using the block
+  header version field as a bitfield.  Bits in the field were used for
+  signaling only---never to reject a block as invalid---and they could
+  be set in parallel.  Measurements were taken only once every 2,016
+  blocks, minimizing the window during which a small percentage of
+  hashrate would need to be lucky to project a false impression of 95%
+  signaling.  Finally, when the 95% signaling threshold was reached,
+  there would be a 2,016 block (roughly two week) "lock in" period until
+  activation occurred to allow people to prepare.  If the signaling
+  threshold wasn't reached by a timeout date, the attempt ended,
+  allowing the unused code to be removed from later node software
+  releases.
+
+  This activation method was first tested with the soft fork that added
+  [BIP68][] consensus-enforced sequence numbers, [BIP112][]
+  `OP_CHECKSEQUENCEVERIFY`, and [BIP113][] nLockTime enforcement by
+  median time past.  The fork quickly made it to the locked-in phase,
+  and then proceeded automatically to activation.
+
+  #### [2016-7] BIP9, BIP148, and BIP91: the BIP141/143 segwit activation
+
+  The segwit soft fork was implemented to use the [BIP9][] activation
+  method.  A few miners quickly began signaling readiness but support
+  peaked far below the needed 95% threshold.  Some Bitcoin users felt
+  miners were illegitimately delaying a useful new feature and supported
+  a mandatory activation mechanism specified in [BIP148][].  The final
+  form of BIP148 required any node following its rules to reject any
+  blocks that didn't signal support for BIPs 141/143 starting on a
+  certain date.
+
+  BIP148 created the risk of a consensus failure.  If miners failed to
+  follow the BIP148 rules and most users continued to consider those
+  miners' blocks as valid, users of BIP148 might accept bitcoins that
+  most other Bitcoin users would consider invalid.  Alternatively, if
+  most users of Bitcoin followed the BIP148 rules---but miners still
+  produced many BIP148-invalid blocks---those users who didn't enforce
+  the BIP148 could accept bitcoins that the BIP148 users would consider
+  invalid.  Only if miners were persuaded to follow the BIP148 rules (or
+  if no users tried to enforce the BIP148 rules) would safety be
+  assured.
+
+  Around the time BIP148 began receiving significant public support,
+  everal miners, exchanges, and other businesses signed their support
+  for a two-step proposal that started with activation of segwit in a
+  way that would follow the BIP148 rules.  This first stage was proposed
```

Comment:
BIP148 required *all* blocks signal support for segwit after August 1st.  If the NYA signers just wanted to get segwit activated, they could've tolerated up to 5% non-signaling blocks, and they also could've taken extra time since the BIP141 deployment parameters for BIP9 didn't time out until November 15th.

Instead, BIP91 specified that, once it activated, all non-signaling blocks would be rejected by miners.  It looks pretty clearly to me like they designed their activation of segwit to specifically not violate any of the BIP148 rules so that there was no risk of a consensus failure between UASF nodes and non-UASF nodes (with miners caught in between, probably losing money and getting yelled at by both sides).